src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/HttpMethodsProvider.cs

```diff
@@ -134,18 +137,130 @@ public HttpMethodsProvider(IHttpContextAccessor httpContextAccessor)
                                  select new JProperty(param.Key, JArray.FromObject(param.Value.ToArray()))).ToArray()))
             };
 
+            // This should be deprecated
             _requestFormAsJsonMethod = new GlobalMethod
             {
                 Name = "requestFormAsJson",
                 Method = serviceProvider => (Func<JObject>)(() =>
                     new JObject((from field in httpContextAccessor.HttpContext.Request.Form
                                  select new JProperty(field.Key, JArray.FromObject(field.Value.ToArray()))).ToArray()))
             };
+
+            _deserializeRequestDataMethod = new GlobalMethod
+            {
+                Name = "deserializeRequestData",
+                Method = serviceProvider => (Func<JObject>)(() =>
+                {
+                    JObject result = null;
+                    var sanitizer = serviceProvider.GetRequiredService<IHtmlSanitizerService>();
```

Comment:
We don't do this anywhere else in controllers, editors, ... We handle that when user data is rendered.

Reply:
Add some validation that will prevent the form from being saved instead if it contains some script. With a custom validation step.

Reply:
Ok removing. It adds overhead to each requests.